### PR TITLE
feat: add support for tailLogEntry API

### LINKED ISSUE #592: feat: Implement user friendly interface for tailLogEntries API
**Is your feature request related to a problem? Please describe.**
It is hard and contra-intuitive to use auto-generated implementation of the grpc or REST APIs. The Java logging client library is missing [tail log entries](https://cloud.google.com/logging/docs/reference/v2/rpc/google.logging.v2#google.logging.v2.LoggingServiceV2.TailLogEntries) implementation.

**Describe the solution you'd like**
I would like to have `tailLogEntries` interface implemented in a similar way as other interfaces (e.g. `listLogEntries`).

**Describe alternatives you've considered**
An alternative would be to use auto-generated API classes such as `com.google.logging.v2.TailLogEntriesRequest`.

**Additional context**
None

### PR DESCRIPTION
**Pull request by minherz**
Implements tailLogEntry API that exposes entries.tail API for end users. The API is in Beta yet and does not support resuming and retrying mechanisms.

Fixes #592

### CODE REVIEW

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntryIterator.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+package com.google.cloud.logging;
+
+import com.google.common.collect.Lists;
+import com.google.logging.v2.TailLogEntriesResponse;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+public class LogEntryIterator implements Iterator<LogEntry> {
```
Comment (simonz130) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntryIterator.java:24]: Document this method. While at first name explains it, but then you look at TailLogEntriesResponse object and without prior context it doesn't look obvious why.

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntryServerStream.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+package com.google.cloud.logging;
+
+import com.google.api.gax.rpc.BidiStream;
+import com.google.common.collect.Lists;
+import com.google.logging.v2.TailLogEntriesRequest;
+import com.google.logging.v2.TailLogEntriesResponse;
+import java.util.Iterator;
+
+public class LogEntryServerStream implements Iterable<LogEntry> {
```
Comment (simonz130) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntryServerStream.java:25]: document the class

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java [hunk 3]
```diff
@@ -919,6 +924,54 @@ static ListLogEntriesRequest listLogEntriesRequest(
     return listLogEntriesAsync(getOptions(), optionMap(options));
   }
 
+  static TailLogEntriesRequest tailLogEntriesRequest(
```
Comment (simonz130) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java:927]: Is this not private for testing purposes?
I suggest you rename it to createTailLogEntriesRequest or buildTailLogEntriesRequest to follow conventions for builder method names.

**Review by simonz130 — APPROVED** [review]
Very solid PR. Thanks @minherz!

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java (hunk 3 above)
Comment (minherz) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java:927]: > Is this not private for testing purposes?
Yes. Similarly to `listLogEntriesRequest()`.
I will rename it. The old habbit to keep the code homogeneous including naming convention dies hard.

**Comment by minherz** [comment]
@simonz130 I will apply all you comments in PR #628 to avoid another review cycle.

**Pull request merged**